File: pyg-manager-web/src/main/webapp/js/service/typeTemplateService.js
// 业务逻辑层和后台数据交互
app.service("typeTemplateService",function ($http) {

    this.findAll = function () {         //创建方法
        // $http.post("../typeTemplate/findAll").success(function (response) { //远程内置服务提交异步请求
        //     $scope.list = response;    //接受后台数据   json--> 数组      调用参数是遍历
        // })
       return $http.post("../typeTemplate/findAll");
    };

    //保存方法
    this.add=function (entity) {
        return $http.post("../typeTemplate/add",entity);
    };
    //修改方法
    this.update=function (entity) {
        return $http.post("../typeTemplate/update",entity);
    };

    //数据回显
    this.findOne = function (id) {
        return $http.get("../typeTemplate/findOne?id=" + id)
    };
    //删除
    this.dele = function (selectIds) {
        return $http.get("../typeTemplate/dele?Ids="+selectIds)
        };

    //模糊查询
    this.query = function (pageNo,pageSize,queryEntity) {
        return $http.post("../typeTemplate/query?pageNo="+pageNo+"&pageSize="+pageSize,queryEntity)
    };

});

File: pyg-manager-interface/src/main/java/com/pyg/managerinterface/TypeTemplateService.java
package com.pyg.managerinterface;



import com.pyg.bean.PageResult;
import com.pyg.bean.ResultMessage;
import com.pyg.pojo.TbTypeTemplate;

import java.util.List;
import java.util.Map;

/**
 * @author Technicolor
 * @version v1.0
 * @date 2019/3/7 21:39
 * @description TODO
 **/


public interface TypeTemplateService {

    /**
     * 查询所有品牌
     * @return
     */
    List<TbTypeTemplate> findAll();

    /**
     * 分页查询
     * @param pageNo
     * @param pageSize
     * @return
     */
    PageResult findPage(Integer pageNo, Integer pageSize);

    /**
     * 新增品牌
     * @param typeTemplate
     * @return
     */
    ResultMessage add(TbTypeTemplate typeTemplate);

    /**
     * 修改数据回现
     * @param id
     * @return
     */
    TbTypeTemplate findOne(Long id);

    /**
     * 根据id修改品牌
     * @param typeTemplate
     * @return
     */
    ResultMessage update(TbTypeTemplate typeTemplate);

    /**
     * 根据id删除
     * @param ids
     * @return
     */
    void dele(Long[] ids);

    /**
     * 模糊分页查询
     * @param pageNo
     * @param pageSize
     * @param typeTemplate
     * @return
     */

    PageResult queryPage(int pageNo, int pageSize, TbTypeTemplate typeTemplate);

    /**
     * 根据模板id返回扩展属性值
     * @param id
     * @return
     */
    List<Map> findSpecList(Long id);
}
